dss/operations/checkout.py

```diff
@@ -0,0 +1,77 @@
+"""
+Tools for managing checkout buckets
+"""
+import os
+import typing
+import argparse
+import logging
+import functools
+import boto3
+
+from dss import Replica
+from dss.operations import dispatch
+from dss.api.bundles import get_bundle_manifest
+from dss.storage.hcablobstore import compose_blob_key
+from cloud_blobstore import BlobNotFoundError
+from cloud_blobstore.gs import GSBlobStore
+from cloud_blobstore.s3 import S3BlobStore
+from google.cloud.storage import Client
+
+
+logger = logging.getLogger(__name__)
+
+checkout = dispatch.target("checkout", help=__doc__)
+
+@functools.lru_cache()
+def get_handle(replica):
+    if "aws" == replica:
+        return S3BlobStore(boto3.client("s3"))
+    elif "gcp" == replica:
+        gcp_client = Client.from_service_account_json(
+            os.environ['GOOGLE_APPLICATION_CREDENTIALS'],
+        )
+        return GSBlobStore(gcp_client)
+    else:
+        msg = f"Unknown replica {replica}"
+        logger.error(msg)
+        raise Exception(msg)
+
+
+def verify_delete(handler, bucket, key):
+    try:
+        size = handler.get_size(bucket, key)
+    except BlobNotFoundError:
+        logger.warning(f'Unable to locate {bucket}/{key} ')
+        return
+    if size:
+        logger.warning(f'attempting removal of key: {key}')
+        handler.delete(bucket, key)
+        try:
+            handler.get(bucket=bucket, key=key)
+        except BlobNotFoundError:
+            logger.warning(f'Success! unable to locate key {key} in bucket: {bucket} ')
+
+
+@checkout.action("remove_bundle_from_checkout",
+                 arguments={"--replica": dict(choices=[r.name for r in Replica], required=True),
+                            "--fqid": dict(nargs="+", help="bundle fqids to remove", required=True)})
+def remove_bundle_from_checkout(argv: typing.List[str], args: argparse.Namespace):
+    """
+    Remove a bundle from the checkout bucket
+    """
+    replica = Replica.aws if args.replica == 'aws' else Replica.gcp
+    handler = get_handle(replica=replica.name)
+    bucket = replica.checkout_bucket
+    for _fqid in args.fqid:
+        uuid, version = _fqid.split('.', 1)
+        manifest = get_bundle_manifest(replica=replica, uuid=uuid, version=version)
```

Comment:
It is not necessary to download the bundle manifest.

This can be verified by investigating the checkout bucket object layout:
```
aws s3 ls s3://org-humancellatlas-dss-checkout-dev/bundles/fec28c45-faf0-454b-a34d-ad4f9f7f3e1a.2019-07-09T150638.531491Z/
2019-07-09 15:07:22   67108865 async_copied_file
2019-07-09 15:07:25        438 cell_suspension_0.json
2019-07-09 15:07:25        561 dissociation_protocol_0.json
2019-07-09 15:07:25        644 donor_organism_0.json
2019-07-09 15:07:22        498 enrichment_protocol_0.json
2019-07-09 15:07:25        607 library_preparation_protocol_0.json
2019-07-09 15:07:22       1702 links.json
2019-07-09 15:07:23        360 process_0.json
2019-07-09 15:07:25        349 process_1.json
2019-07-09 15:07:25        349 process_2.json
2019-07-09 15:07:25       3312 project_0.json
2019-07-09 15:07:22        419 sequence_file_0.json
2019-07-09 15:07:25        419 sequence_file_1.json
2019-07-09 15:07:25        586 sequencing_protocol_0.json
2019-07-09 15:07:25        811 specimen_from_organism_0.json
```